pkgs/development/libraries/geogram/default.nix

```
@@ -0,0 +1,91 @@
+{ lib
+, stdenv
+, cmake
+, fetchFromGitHub
+, python3Packages
+}:
+
+let
+  testdata = fetchFromGitHub {
+    owner = "BrunoLevy";
+    repo = "geogram.data";
+    rev = "70c51a4968960e2d4fa652421b5721672f2b2b70";
+    hash = "sha256-vJtbDDjt2t/6TH0k3i287yGZRd1+qbw20F8D8CFgb3c=";
+  };
+in
+stdenv.mkDerivation rec {
+  pname = "geogram";
+  version = "1.8.2";
+
+  src = fetchFromGitHub {
+    owner = "BrunoLevy";
+    repo = pname;
+    rev = "v${version}";
+    hash = "sha256-8VpBPtFMSDr5Al72ckRxHjoM4vOCzMl52MlEYyG+MSs=";
+    fetchSubmodules = true;
+  };
+
+  outputs = [ "bin" "lib" "dev" "out" ];
+
+  cmakeFlags = [
+    # Triangle has unclear licensing situation
```

it's very clearly "unfree" in our context

https://github.com/BrunoLevy/geogram/wiki/ThirdParty#geometry

> [triangle (optional)](https://www.cs.cmu.edu/~quake/triangle.html), license: "Although Triangle is freely available, it is copyrighted by the author and may not be sold or included in commercial products without a license" 2D constrained Delaunay tiangulation.

Can you make it optional and when enabled, add "unfree" license to the package?

Not sure if it's worth this effort tho.
```suggestion
    # Triangle has unfree license, see https://www.cs.cmu.edu/~quake/triangle.html
```

---

I think I'd just leave it for the moment, since I've also disabled a few other package options. I'm packaging GeoGram for Meshroom, and AFAIK Meshroom doesn't need Triangle.

Regarding the other options:
* If `GEOGRAM_WITH_LEGACY_NUMERICS` is enabled GeoGram will build its own version of ARPACK, CBLAS, CLAPACK, LIBF2C and SUPERLU, which seems wasteful. Everything still seems to work when turning it off, so I just did that.
* `GEOGRAM_WITH_LUA`: Same story, GeoGram will build its own Lua interpreter. Meshroom doesn't need it so I turned it off.
* `GEOGRAM_WITH_GRAPHICS`: Pulls in a vendored version GLFW (or a system version if another flag is set). Again, I'm not really sure what it's used for, but Meshroom works without it so I turned it off.

I figured it's a good idea to drop the compile time a little by disabling nonessential flags.

---

Sounds reasonable. Can you add that in a comment in the code so it can be later understood?

---

Sure, will do